packages/cdktf/lib/terraform-element.ts

```diff
@@ -76,12 +76,10 @@ export class TerraformElement extends Construct {
   private _friendlyUniqueId?: string;
 
   public get friendlyUniqueId() {
-    if (!this._friendlyUniqueId) {
-      if (this._logicalIdOverride) {
-        this._friendlyUniqueId = this._logicalIdOverride;
-      } else {
-        this._friendlyUniqueId = this.cdktfStack.getLogicalId(this);
-      }
+    if (this._logicalIdOverride) {
+      this._friendlyUniqueId = this._logicalIdOverride;
+    } else if (!this._friendlyUniqueId) {
+      this._friendlyUniqueId = this.cdktfStack.getLogicalId(this);
     }
```

Comment:
To my understanding of what it should be doing, it wasn't working.

So in the new `renameResourceId` function we call `this.overrideLogicalId` to set the new id associated with the rename. Which just adds to a global on TerraformElement:

```ts
public overrideLogicalId(newLogicalId: string) {
    ok(
      !this._fqnToken,
      "Logical ID may not be overridden once .fqn has been requested. Make sure to override the id before passing the construct to other constructs."
    );
    this._logicalIdOverride = newLogicalId;
  }
```

Before those changes to `friendlyUniqueId()` of TerraformElement, the new logical id set by `this.overrideLogicalId` didn't come through in calling `this.friendlyUniqueId` on the instance of the resource as `this.friendlyUniqueId` was already set. 

As `friendlyUniqueId()` was before:
```ts
public get friendlyUniqueId() {
    if (!this._friendlyUniqueId) {
      if (this._logicalIdOverride) {
        this._friendlyUniqueId = this._logicalIdOverride;
      } else {
        this._friendlyUniqueId = this.cdktfStack.getLogicalId(this);
      }
    }
```

Figured it just wasn't working as it was suppose to as I couldn't figure out how the id override could otherwise come through. And I'd imagine we'd still want to be able to override it even it is does currently have an id.

Though correct me if I'm wrong 😅

Reply:
A safer route could be to make a separate private function that does this, might avoid adding a potential footgun.

Reply:
Descoped `renameResourceId`, related changes to id behavior has been reverted.